### PR TITLE
Change RealmReact to static library and support rnpm

### PR DESCRIPTION
**Pull request by appden**
We needed to do this due to the way React Native modules work. Our hacky solution for using a framework ended up falling apart due to it relying on symbols not getting stripped from the main application.

This static library is in its own Xcode project because rnpm links in *all* static libraries from a given Xcode project.

The postlink script takes care of linking in libc++ and libz. So now the only setup instructions will be: `rnpm link realm`

### CODE REVIEW

**Comment by alazier** [comment]
Please make the rnpm stuff a separate pr. Not sure we need that right now and it will incur more testing which would be nice to not have to do asap.

**Comment by alazier** [comment]
Can we have our other static libraries in subprojects? or is that what we are doing?

**Comment by alazier** [comment]
I guess we can leave the rnmp stuff for now and not push a build until we are ready.

**Comment by alazier** [comment]
I think ReactNative.xcodeproj should be at the root level.

**Comment by appden** [comment]
All of our Android stuff lives under `react-native/android` so I think it makes the most sense to do the same for iOS

File: scripts/rnpm-postlink.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        var mainGroup = project.getFirstProject().firstProject.mainGroup;
+
+        project.getPBXGroupByKey(mainGroup).children.push({
+            value: group,
+            comment: 'Frameworks',
+        });
+    }
+
+    ['libc++', 'libz'].forEach(function(name) {
+        project.addFramework('usr/lib/' + name + '.tbd', {
```
Comment (alazier) [inline scripts/rnpm-postlink.js]: what if the framework is already there?

**Comment by alazier** [comment]
Where does our framework get added to the user's project?

File: scripts/rnpm-postlink.js (hunk 1 above)
Comment (appden) [inline scripts/rnpm-postlink.js]: It's a no-op. I confirmed both in testing and in the code.

**Comment by appden** [comment]
`rnpm` does that automatically for every static library in the project

**Comment by alazier** [comment]
I think once we add a test to make sure `rnpm link realm` works we can :+1:

**Pull request merged**